starfish/image/_filter/scale_by_percentile.py

```diff
@@ -54,16 +55,15 @@ def scale(image: np.ndarray, p: int) -> np.ndarray:
         """
         v = np.percentile(image, p)
 
-        # asking for a float percentile clipping value from an integer image will
-        # convert to float, so store the dtype so it can be restored
-        dtype = image.dtype
         image = image / v
-        return image.astype(dtype)
+        image = preserve_float_range(image)
```

Comment:
why is this necessary?

Reply:
if you scale by anything other than the 100th percentile (excluding ties), it will generate values above 1: 

e.g. if the top value is `0.96` and you scale by the 90th percentile, which for this example say is `0.87`, you'll get `1.10`.